run/rails/Dockerfile

```diff
@@ -7,23 +7,38 @@
 FROM ruby:3.2-buster
 # [END cloudrun_rails_base_image]
 
-RUN (curl -sS https://deb.nodesource.com/gpgkey/nodesource.gpg.key | gpg --dearmor | apt-key add -) && \
-    echo "deb https://deb.nodesource.com/node_20.x buster main"      > /etc/apt/sources.list.d/nodesource.list && \
-    apt-get update && apt-get install -y nodejs lsb-release
+# Replace expired Buster repositories with archive.debian.org
+RUN sed -i 's|http://deb.debian.org/debian|http://archive.debian.org/debian|g' /etc/apt/sources.list && \
+sed -i 's|http://security.debian.org/debian-security|http://archive.debian.org/debian-security|g' /etc/apt/sources.list && \
+echo 'Acquire::Check-Valid-Until "false";' > /etc/apt/apt.conf.d/10no-check-valid-until && \
+apt-get update
+
+# Install Node.js (from NodeSource)
+RUN curl -fsSL https://deb.nodesource.com/gpgkey/nodesource.gpg.key | \
+gpg --dearmor -o /usr/share/keyrings/nodesource.gpg && \
+echo "deb [signed-by=/usr/share/keyrings/nodesource.gpg] https://deb.nodesource.com/node_20.x buster main" > \
+/etc/apt/sources.list.d/nodesource.list && \
+apt-get update && \
+apt-get install -y nodejs lsb-release
```

Comment:
nit: it would make more sense to install the lsb-release package in the same step that the repository is setup